lists-service/cicd/base/app/base.yaml

```diff
@@ -69,7 +75,40 @@ Resources:
                    "db-password" : ""
                 }
 
-
+  AppClient:
+    Type: AWS::CloudFormation::Stack
+    Properties:
+      Parameters:
+        pName: !Sub
+          - species-lists-${ResourceName}
+          - ResourceName: !If [ IsDev, !Ref pCleanBranch, !Ref pEnvironment ]
+        pCallbackUrls: !Join
+          - ','
+          - - !Sub https://${pAppClientDomain}
+        pDefaultUrl: !Sub https://${pAppClientDomain}
+        pLogoutUrls: !Join
+          - ','
+          - - !Sub https://${pAppClientDomain}
+        pScopes: !Join
+          - ','
+          - - email
+            - openid
+            - profile
+            - ala/roles
+            - ala/attrs
+        pGenerateSecret: false
+        pAllowedOAuthFlows: !Join
+          - ','
+          - - code
+        pUserPoolId:
+          Fn::ImportValue:
+            Fn::Sub: ${pCognitoStackName}-UserPoolId
+      TemplateURL: !Sub
+        - https://${AppClientBucket}.s3.${AWS::Region}.amazonaws.com/cicd/cognito/app/app_client.yaml
+        - AppClientBucket:
+            Fn::ImportValue:
+              Fn::Sub: ${pCognitoStackName}-AppClientBucket
+      TimeoutInMinutes: 5
```

Comment:
This seems like it belongs in the deploy component rather than base as it's only relevant to the application. Then the pAppClientDomain can be got from the existing sub_domain / hosted_zone thats already available in that component

Reply:
The setup of the app client actually needs to be setup in a shared base stack when we merge the UI and service components, the `oidcClientId` is required by both components. 
I put it in the lists service base stack to keep it out of deploy component until the merge is done.